plansys2_msgs/msg/Knownledge.msg

```diff
@@ -0,0 +1,3 @@
+string[] instances
+string[] predicates
+string goal
```

Comment:
@fmrico 
Isn't the `functions` valuations part of the knowledge ?

Reply:
Yes, absolutely. Please, add it too.

Maybe we should add an example in the examples repo. I didn't add functions because the others elements were not in those examples.
Anyway, it was available, but not in plansys2 namespace.

Thanks to be aware of this.

Reply:
Ok, I just update to this the current version of the master, and the functions do not seems to be functional anymore. I try to figure out what append and I will create an issue.

Reply:
:thinking: I merged #82 , which included a lot of improvements. I ran the tests and tested https://github.com/IntelligentRoboticsLabs/ros2_planning_system_examples, and everything seemed to be working.  I think that this PR could affect this. Did you add tests for `functions`? I remember that there were some of them...

Reply:
In my application, I had issues that I finally manage to solve:
- The name of the service to manage function changed, but I found it quite easily
- A service "Set function" and a service "update function" have been created, and it took me some time to understand that the "update function" fails if the function have not been "set" before. Is this useful maintain two different service ?
- A last point is described in #87, and took me some time to understand what append.

I still have some performance issue in launching the execution of the plan. The start of plan execution take much longer than before. I'm currently trying to identify where those slowness came from.

Reply:
Did you find the performance issue?